Lib/nturl2path.py

```diff
@@ -48,16 +49,13 @@ def pathname2url(p):
         p = p[4:]
         if p[:4].upper() == 'UNC/':
             p = '//' + p[4:]
-        elif p[1:2] != ':':
-            raise OSError('Bad path: ' + p)
-    if not ':' in p:
-        # No DOS drive specified, just quote the pathname
-        return urllib.parse.quote(p)
-    comp = p.split(':', maxsplit=2)
-    if len(comp) != 2 or len(comp[0]) > 1:
-        error = 'Bad path: ' + p
-        raise OSError(error)
+    drive, tail = ntpath.splitdrive(p)
+    if drive[1:2] == ':':
```

Comment:
We shouldn't ever get more than `x:` here, right? Can we check `drive[1:]` instead so that we don't one day discover that people can smuggle unescaped characters in here?

```suggestion
    if drive[1:] == ':':
```